### PR TITLE
Fixes lint problems reported by Clippy in Rust 1.80.0

### PR DESCRIPTION
**Pull request by Superhepper**
- In Rust 1.80.0 the `size_of` function is added to the preamble which causes lint errors when used with full qualifiers in the code so for that specific function the lint error has been suppressed.

- Other unused qualifiers and documentation lint errors have been fixed.

- Fixed redundant code in build.rs

- Fixed a configuration flag not being used correctly.

### CODE REVIEW

File: tss-esapi/src/structures/nv/storage/public.rs
Comment (ionut-arm) [inline tss-esapi/src/structures/nv/storage/public.rs:30]: Does the compiler complain if you remove the "unused" qualifications?

**Review by ionut-arm — APPROVED** [review]
Thanks!

File: tss-esapi/src/structures/nv/storage/public.rs
Comment (Superhepper) [reply tss-esapi/src/structures/nv/storage/public.rs:30]: I am not sure. The call is specified using the full qualifications but because 1.80 puts the size_of in the preamble the qualifications do not get used and that is what it complains about.

I will test it.

**Comment by tmfink** [comment]
I need these changes to build with Rust 1.80.0

**Comment by Superhepper** [comment]
Yes, we just need to find another reviewer to have a look at it and approve it before it can be merged.

**Review by wiktor-k — APPROVED** [review]
It's okay for me too but if you can check the thing that Ionut asked for it'd be 👌

File: tss-esapi/src/structures/nv/storage/public.rs
Comment (Superhepper) [reply tss-esapi/src/structures/nv/storage/public.rs:30]: You cannot remove it completely because then you will get an error.
And there is no 'qualifications' lint so you cannot write ```#[allow(qualifications)]```.

**Pull request merged**
